File: factors.py
#!/usr/bin/python3
#import pdb; pdb.set_trace()

import math
import sys

def factorize(number):
    factors = []
    divisor = 2

    while divisor <= math.isqrt(number):
        if number % divisor == 0:
            factors.append(divisor)
            number //= divisor
        else:
            divisor += 1

    if number > 1:
        factors.append(number)

    return factors

def main(file_path):
    with open(file_path, 'r') as file:
        numbers = [int(line.strip()) for line in file]

    for number in numbers:
        factors = factorize(number)
        print(f"{number}={'*'.join(map(str, factors))}")

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: factors <file>")
        sys.exit(1)

    file_path = sys.argv[1]
    
    try:
        main(file_path)
    except Exception as e:
        print(f"Error: {e}")
        sys.exit(1)

File: factors.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(int argc, char *argv[]) {
    FILE *file;
    char *line = NULL;
    size_t len = 0;
    ssize_t read;

    if (argc != 2) {
        printf("Usage: factors <file>\n");
        return 1;
    }

    file = fopen(argv[1], "r");
    if (file == NULL) {
        printf("Error opening file.\n");
        return 1;
    }

    while ((read = getline(&line, &len, file)) != -1) {
        long int number = atol(line);
        printf("%li=", number);

        int factor = 2;
        while (factor <= sqrt(number)) {
            if (number % factor == 0) {
                printf("%i*", factor);
                number = number / factor;
            } else {
                factor++;
            }
        }
        
        printf("%li\n", number);
    }

    fclose(file);
    if (line) {
        free(line);
    }

    return 0;
}
